File: s3/ruby/s3_client.rb
#!/usr/bin/env ruby
require 'aws'

AWS.config({
  :access_key_id => ENV['AWS_ACCESS_KEY_ID'],
  :secret_access_key => ENV['AWS_SECRET_ACCESS_KEY'],
  :region => ENV['AWS_REGION']
})

s3 = AWS::S3.new

bucket = s3.buckets['your-bucket']
obj = bucket.objects['key']
obj.write(Pathname.new('/path/to/file.txt'))

obj.read do |chunk|
  puts chunk
  #=> data of s3://your-bucket/key
end

File: s3/nodejs/s3_client.js
var AWS = require('aws-sdk');
var fs = require('fs');

AWS.config.loadFromPath('/path/of/your/config.json');

var s3 = new AWS.S3();

var bodyStream = fs.createReadStream( '/path/to/file.txt' );
var params = {
  Bucket: 'your-bucket',
  Key: 'key',
  Body: bodyStream
};

s3.putObject(params, function(err, data) {
  if (err)
    console.log(err);
  else
    console.log("Successfully uploaded.");
});


var params = {
  Bucket: 'your-bucket',
  Key: 'key'
};

s3.getObject(params, function(err, data) {
  if (err)
    console.log(err);
  else
    console.log(data.Body.toString());
});

